packages/menu/src/Menu.tsx

```diff
@@ -395,14 +393,12 @@ function Menu({
           trigger.props.onClick(e);
         }
       },
-      children: triggerChildren
-        ? [
-            ...(triggerChildren instanceof Array
-              ? triggerChildren
-              : [triggerChildren]),
-            popoverContent,
-          ]
-        : popoverContent,
+      children: (
+        <>
+          {trigger.props.children}
```

Comment:
Hmm i'm not sure I understand whats going on with the old bit of code but this new solution is the same approach that the tooltip uses.

Was the old code passing the children as an array and thats why a key was needed?